src/com/company/IntListImpl.java

```diff
@@ -0,0 +1,220 @@
+        } else {
+            array[index] = item;
+        }
+        return item;
+    }
+
+    @Override
+    public Integer removeIt(Integer item) {
+        for (int i = 0; i < array.length; i++) {
+            if (array[i] == item) {
+                array[i] = 0;
+                size--;
+                return Integer.parseInt(item + ": Удален");
+            }
+        }
+        throw new NullElementInTheArray(item + ": данный объект не найден");
+    }
+
+    @Override
+    public Integer remove(int index) {
+        int number;
+        if (array[index] == 0) {
+            throw new NullElementInTheArray("В ячейке под индесом " + index + " null");
+        } else {
+            number = array[index];
+            array[index] = 0;
+            size--;
+        }
+        return Integer.parseInt("Элемент: " + number + " под индексом " + index + " удален");
+    }
+
+    @Override
+    public boolean contains(Integer item) {
+        Integer[] arr = array;
+        sort(arr);
+        Integer element = searchBin(array, item);
+        if (element != 0) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    @Override
+    public Integer indexOf(Integer item) {
+        for (int i = 0; i < array.length; i++) {
+            if (array[i] == item) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    @Override
+    public Integer lastIndexOf(Integer item) {
+        for (int i = array.length; i >= 0; i--) {
+            if (array[i - 1] == item) {
+                return i - 1;
+            }
+        }
+        return -1;
+    }
+
+    @Override
+    public Integer get(int index){
+        if (array[index] == 0) {
+            throw new NullElementInTheArray("В ячейке под индексом " + index + " объект отсутствует");
+        } else {
+            return array[index];
+        }
+    }
+
+    @Override
+    public boolean equals(IntList otherList) {
+        boolean result = true;
+        if (otherList == null) {
+            throw new NullElementInTheArray("Один из объектов отсутствует!");
+        } else if (array.length != otherList.size()) {
+            return false;
+        } else {
+            int eL = 0;
+            for (int i = 0; i < array.length; i++) {
+                if (array[i] == otherList.get(i)) {
+                    eL++;
+                }
+            }
+            if (eL == size) {
+                return result;
+            }
+        }
+        return result;
+    }
+
+    @Override
+    public int size() {
+        return size;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return size() == 0;
+    }
+
+    @Override
+    public void clear() {
+        for (int i = 0; i < array.length; i++) {
+            array[i] = 0;
+        }
+        size = 0;
+    }
+
+    @Override
+    public Integer[] toArray() {
+        Integer[] array = new Integer[10];
+        return array;
```

Comment:
метод должен вернуть исходные данные в виде массива, а не создавать новый
```suggestion
        return Arrays.copyOf(array, size);
```